client/src/components/accordionMenu.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactNode, useEffect, useRef, useState } from "react";
+
+type PropsType = {
+  name: string;
+  contents?: string[];
+};
+
+export const AccordionMenu = ({ name, contents }: PropsType) => {
+  const childElement = useRef<HTMLDivElement>(null);
+  const [showChildren, setShowChildren] = useState(false);
+  const [childHeight, setChildHeight] = useState(0);
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    if (childElement.current) {
+      const height = childElement.current?.clientHeight;
+      setChildHeight(height);
+    }
+  }, []);
+
+  const handleClick = () => {
+    if (childElement.current) {
+      setIsOpen(true);
+      if (isOpen == true) {
+        setIsOpen(false);
+      }
```

Comment:
```suggestion
 setIsOpen((prev) => !prev);
```
これで、現在の変数をもとに not現在の値みたいにできる。